### PR TITLE
add mariadb-connector-c/3.1.11

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **mariadb-connector-c/3.1.11**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

closes https://github.com/conan-io/conan-center-index/issues/1695

### CODE REVIEW

File: recipes/mariadb-connector-c/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            del self.options.with_iconv
+            self.options.with_ssl = "schannel"
```
Comment (madebr) [inline recipes/mariadb-connector-c/all/conanfile.py:44]: You're assigning `schannel` to `self.otions.with_ssl`.
So I'm assuming it's ok to use another ssl library on Windows?

File: recipes/mariadb-connector-c/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    def package(self):
+        self.copy("COPYING.LIB", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "symbols"))
+
+    def package_info(self):
+        self.cpp_info.names["pkg_config"] = "libmariadb"
+        self.cpp_info.includedirs = [os.path.join("include", "mariadb")]
```
Comment (madebr) [inline recipes/mariadb-connector-c/all/conanfile.py:123]: I would append the `mariadb` include directory.
When installing in `/usr` (by the system package manager), it is also possible to do `#include <mariadb/xxx>`.

File: recipes/mariadb-connector-c/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            self._cmake.definitions["WITH_MYSQLCOMPAT"] = False
+            self._cmake.definitions["WITH_ICONV"] = self.options.with_iconv
+        self._cmake.definitions["WITH_UNIT_TESTS"] = False
+        self._cmake.definitions["WITH_DYNCOL"] = self.options.dyncol
+        self._cmake.definitions["WITH_EXTERNAL_ZLIB"] = True
+        self._cmake.definitions["WITH_CURL"] = self.options.with_curl
+        self._cmake.definitions["WITH_SSL"] = self.options.with_ssl
+        self._cmake.definitions["INSTALL_BINDIR"] = os.path.join(self.package_folder, "bin")
+        self._cmake.definitions["INSTALL_LIBDIR"] = os.path.join(self.package_folder, "lib")
+        self._cmake.definitions["INSTALL_PLUGINDIR"] = os.path.join(self.package_folder, "lib", "plugin")
```
Comment (madebr) [inline recipes/mariadb-connector-c/all/conanfile.py:105]: I think the MSVC error is caused by the backspace escaping characters.
This means that some build path might get embedded in the executable (e.g. the plugins).
So this library maybe requires some environment variables set.

```suggestion
        self._cmake.definitions["INSTALL_BINDIR"] = os.path.join(self.package_folder, "bin").replace("\\", "/")
        self._cmake.definitions["INSTALL_LIBDIR"] = os.path.join(self.package_folder, "lib").replace("\\", "/")
        self._cmake.definitions["INSTALL_PLUGINDIR"] = os.path.join(self.package_folder, "lib", "plugin").replace("\\", "/")
```

File: recipes/mariadb-connector-c/all/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/mariadb-connector-c/all/conanfile.py:44]: Yes it's ok to use others ssl libraries on windows. It's just a conditional default option.

upstream default `WITH_SSL` value:
- windows: SCHANNEL
- non windows: OPENSSL

https://github.com/mariadb-corporation/mariadb-connector-c/blob/79137a4ae1cf37ab46940d26879051ad1dfce512/CMakeLists.txt#L67
https://github.com/mariadb-corporation/mariadb-connector-c/blob/79137a4ae1cf37ab46940d26879051ad1dfce512/CMakeLists.txt#L267

Comment (madebr) [reply recipes/mariadb-connector-c/all/conanfile.py:44]: Thans for the links.

File: recipes/mariadb-connector-c/all/conanfile.py (hunk 3 above)
Comment (SpaceIm) [reply recipes/mariadb-connector-c/all/conanfile.py:105]: On the other hand, it fails with Visual Studio 16 only (on my computer it doesn't fail, but long path is enabled at system level). Maybe just add `short_paths`?
Regarding environment variables, yes maybe, need to dig a little bit more.

File: recipes/mariadb-connector-c/all/conanfile.py (hunk 2 above)
Comment (SpaceIm) [reply recipes/mariadb-connector-c/all/conanfile.py]: done

**Review by madebr — APPROVED** [review]
Upstreaming some of the patches (in modified form) would be nice

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**